corehq/messaging/smsbackends/telerivet/views.py

```diff
@@ -60,14 +60,21 @@ def message_status(request, message_id):
     status = request.POST.get('status')
     logger.info(f'Updating Telerivet message status: message_id={message_id}, status={status}')
 
-    process_message_status.delay(
-        message_id,
-        status,
-        request.POST.get('secret', ''),
-        error_message=request.POST.get('error_message', ''),
-    )
+    backend = SQLTelerivetBackend.by_webhook_secret(request.POST.get('secret', ''))
```

Comment:
`by_webhook_secret` would fire a query even when `secret` is not there, so can we have this fired only `if request.POST.get('secret', '')`

Reply:
Agree ... unless "" is a valid secret. (?!)

If `by_webhook_secret()` is going to get called a lot, you may want to cache its return value.

You can also speed it up by changing it to ...
```python
    @classmethod
    def by_webhook_secret(cls, webhook_secret):
        # ...
        result = cls.active_objects.filter(
            hq_api_id=cls.get_api_id()
        )
        for backend in result:
            if backend.config.webhook_secret == webhook_secret:
                return backend
```
... which should, on average, halve the number of records to iterate through. (Not your code, I know.)

Reply:
That's true. Especially for bulk broadcasts...

Reply:
@kaapstorm Added caching and little optimisation...